db/db_impl.cc

```diff
@@ -552,6 +552,10 @@ Status DBImpl::SetDBOptions(
         MaybeScheduleFlushOrCompaction();
       }
 
+      if(new_options.wal_bytes_per_sync !=
```

Comment:
You need to do this after mutable_db_options_ is set and inside write thread (between write_thread_.EnterUnbatched() && write_thread_.ExitUnbatched(). Can you merge the logic with those at line 567-574?